socorro/cron/jobs/daily_url.py

```diff
@@ -2,103 +2,334 @@
+        if i == 14:  # bug_associations.bug_id
+            x = ','.join(str(bugid) for bugid in x)
+        if i == 15:  # r.user_comments
+            x = x.replace('\t', ' ')  # per bug 519703
+        if i == 17:  # r.email -- show 'email' if the email is likely useful
+            # per bugs 529431/519703
+            if '@' in x:
+                x = 'yes'
+            else:
+                x = ''
+        if type(x) == str:
+            x = x.strip().replace('\r', '').replace('\n', ' | ')
+        column_value_list.append(x)
+    return column_value_list
+
+
+#------------------------------------------------------------------------------
+def write_row(file_handles_tuple,
+              crash_list):
+    """
+    Write a row to each file: Seen by internal users (full details), and
+    external users (bowdlerized)
+    """
+    private_file_handle, public_file_handle = file_handles_tuple
+    # config.logger.debug("Writing crash %s (%s)",crash_list,len(crash_list))
+    private_file_handle.writerow(crash_list)
+    crash_list[1] = 'URL (removed)'  # remove url
+    crash_list[17] = ''  # remove email
+    if public_file_handle:
+        public_file_handle.writerow(crash_list)
 
+
+#------------------------------------------------------------------------------
+def dailyUrlDump(
+    connection,
+    config,
+    day,
+    get_output_context=gzipped_csv_files,
+    write_row=write_row,
+    process_crash=process_crash
+):
+
+    try:
+        with get_output_context(config, day) as output_context:
+            execute_no_results(
+                connection,
+                """ SET TEMP_BUFFERS = %s """,
+                (config.database_temp_buffer_size,)
+            )
+
+            sql_parameters = setup_query_parameters(config, day)
+            config.logger.debug(
+                "day = %s; now = %s; yesterday = %s",
+                day,
+                sql_parameters.now_str,
+                sql_parameters.yesterday_str
+            )
+
+            with connection.cursor() as db_cursor:
+                sql_query = sql % sql_parameters
+                config.logger.debug("SQL is: %s", sql_query)
+                db_cursor.execute(sql_query)
+                headers_not_yet_written = True
+                while True:
+                    crash_row = db_cursor.fetchone()
+                    if crash_row is None:
+                        break
+                    if headers_not_yet_written:
+                        write_row(
+                            output_context,
+                            [x[0] for x in db_cursor.description]
+                        )
+                        headers_not_yet_written = False
+                    column_value_list = process_crash(crash_row)
+                    write_row(output_context, column_value_list)
+    except Exception, x:
+        config.logger.error(
+            'major problem running dailyUrlDump',
+            exc_info=True
+        )
+        raise
```

Comment:
I don't see the point of the lines above. If there is an exception, crontabber will spit out it to logging and it will also put the full exception into state.